packages/js-client-grpc/src/qdrant-client.ts

```diff
@@ -47,7 +47,9 @@ export class QdrantClient {
             }
             const parsedUrl = new URL(url);
             this._host = parsedUrl.hostname;
-            this._port = parsedUrl.port ? Number(parsedUrl.port) : port;
+            const getPort = (url: string): number | null =>
+                url.includes(':443') ? 443 : url.includes(':80') ? 80 : port;
```

Comment:
In that case parsedUrl.port on line 52 will not return null, so `getPort` will not be invoked.

Reply:
ok, in which case `getPort` will be invoked? If port is not specified explicitly? If so, why are we doing those checks for `url.includes`?

Reply:
Invoked in case of urls having default ports (443 for https and 80 for http) or no port specified.
https://github.com/qdrant/qdrant-js/pull/61#issuecomment-1984027765